src/distributed_join.cuh

```diff
@@ -152,6 +163,219 @@ void all_to_all_comm(cudf::table_view input,
   }
 }
 
+struct compression_functor {
+  /**
+   * Compress a buffer using cascaded compression.
+   *
+   * @param[in] uncompressed_data Input buffer to be compressed.
+   * @param[in] uncompressed_count Number of elements to be compressed. Note that in general this
+   * is different from the size of the buffer.
+   * @param[out] compressed_data Output data after cascaded compression. This argument does not
+   * need to be preallocated.
+   * @param[out] compressed_size Number of bytes of *compressed_data*.
+   */
+  template <typename T>
+  void operator()(const void *uncompressed_data,
+                  size_t uncompressed_count,
+                  rmm::device_buffer &compressed_data,
+                  size_t &compressed_size)
+  {
+    nvcomp::CascadedCompressor<T> compressor(
+      static_cast<const T *>(uncompressed_data), uncompressed_count, 1, 1, true);
```

Comment:
In nvcomp 1.2 we have an automatic selector for cascaded schemes, let's open an issue to track nvcomp 1.2 integration and that should be mostly about moving to the automatic selector.

Reply:
The issue is raised in the internal repo.